src/availability.rs

```diff
@@ -363,10 +366,107 @@ where
                 Ok(result)
             }
             .boxed()
+        })?
+        .get("get_transaction_range_rev", move |req, state| {
+            async move {
+                let height: usize = req.integer_param("height")?;
+                let offset: usize = req.integer_param("offset")?;
+                let limit: usize = req.integer_param("limit")?;
+                let batch_size = NonZeroUsize::new(10).unwrap();
+
+                let step1 = build_range_stream_rev(height, batch_size);
+                let step2 = step1.flat_map(|range| {
+                    expand_range_to_block_range_stream::<State, Types>(range, state, timeout)
+                });
+
+                let step3 = step2.flat_map(|result| {
+                    futures::stream::iter(match result {
+                        Ok(block) => expand_block_data_to_transaction_vec(block)
+                            .into_iter()
+                            .rev()
+                            .map(Some)
+                            .collect(),
+
+                        _ => vec![None],
```

Comment:
I think I have convinced myself that we should just fail the whole endpoint if any blocks are missing, rather than inserting placeholders. The placeholders make it inconsistent with other endpoints in the availability API, plus, since we are querying for the blocks by number, we will always actively fetch them, and thus missing blocks should be resolved very quickly (within seconds), so I'm not sure the placeholders are really doing much besides complicating things.

If we really want the placeholders for the UI in the future, we could create a separate endpoint in a dedicated block explorer server